src/test/java/com/crud/tasks/trello/client/TrelloClientTest.java

```diff
@@ -0,0 +1,100 @@
+    private RestTemplate restTemplate;
+
+    @Mock
+    private TrelloConfig trelloConfig;
+
+    @Test
+    public void shouldFetchTrelloBoards() throws URISyntaxException {
+        //Given
+        lenient().when(trelloConfig.getTrelloApiEndpoint()).thenReturn("http://test.com");
+        lenient().when(trelloConfig.getTrelloAppKey()).thenReturn("test");
+        lenient().when(trelloConfig.getTrelloToken()).thenReturn("test");
+
+        TrelloBoardDto[] trelloBoards = new TrelloBoardDto[1];
+        trelloBoards[0] = new TrelloBoardDto("Kodilla", "test_id", new ArrayList<>());
+
+        URI uri = new URI("http://test.com/members/test/boards?key=test&token=test&fields=name,id&lists=all");
+
+        lenient().when(restTemplate.getForObject(uri,TrelloBoardDto[].class)).thenReturn(trelloBoards);
+        //When
+        List<TrelloBoardDto> fetchedTrelloBoards = trelloClient.getTrelloBoards();
+        //Then
+        assertEquals(1, fetchedTrelloBoards.size());
+        assertEquals("test_id", fetchedTrelloBoards.get(0).getId());
+        assertEquals("Kodilla", fetchedTrelloBoards.get(0).getName());
+        assertEquals(new ArrayList<>(), fetchedTrelloBoards.get(0).getLists());
+    }
+
+    @Test
+    public void shouldCreateCard() throws URISyntaxException{
+        //Given
+        lenient().when(trelloConfig.getTrelloApiEndpoint()).thenReturn("http://test.com");
+        lenient().when(trelloConfig.getTrelloAppKey()).thenReturn("test");
+        lenient().when(trelloConfig.getTrelloToken()).thenReturn("test");
+        TrelloCardDto trelloCardDto = new TrelloCardDto(
+                "Test task",
+                "Test description",
+                "top",
+                "test_id"
+        );
+        URI uri = new URI("http://test.com/cards?key=test&token=test&name=Test%20task&desc=Test%20Description&pos=top&idList=test_id");
+        CreatedTrelloCard createdTrelloCard = new CreatedTrelloCard(
+                "1",
+                "test task",
+                "http://test.com"
+        );
+        lenient().when(restTemplate.postForObject(uri, null, CreatedTrelloCard.class)).thenReturn(createdTrelloCard);
+        //When
+        CreatedTrelloCard newCard = trelloClient.createNewCard(trelloCardDto);
+        //Then
+        assertEquals("1", newCard.getId());
+        assertEquals("test task", newCard.getName());
+        assertEquals("http://test.com", newCard.getShortUrl());
+    }
+
+    @Test
+    public void shouldReturnEmptyList() throws URISyntaxException {
+        //Given
+        lenient().when(trelloConfig.getTrelloApiEndpoint()).thenReturn("http://test.com");
+        lenient().when(trelloConfig.getTrelloAppKey()).thenReturn("test");
+        lenient().when(trelloConfig.getTrelloToken()).thenReturn("test");
+
+        URI uri = new URI("http://test.com/members/test/boards?key=test&token=test&fields=name,id&lists=all");
+
+        lenient().when(restTemplate.getForObject(uri,TrelloBoardDto.class)).thenReturn(null);
```

Comment:
lenient() - sa do wyrzucenia. Znacza ze mock moze zostac nie uzyty. 

Jesli jest uzyty, to nie potrzebuje lenient(), jesli nie jest to jest do wyrzucenia